### PR TITLE
allow empty json cards

### PR DESCRIPTION
**Pull request by xithan**

### CODE REVIEW

File: card/mod/bootstrap/set/abstract/bootswatch_theme.rb [hunk 1]
```diff
@@ -45,13 +45,13 @@
...
 PRE_VARIABLES_CARD_NAMES = %i[
   style_jquery_ui_smoothness
-  style_cards
   style_right_sidebar
   bootstrap_functions
 ].freeze
 
 POST_VARIABLES_CARD_NAMES = %i[
   bootstrap_core
+  style_cards
```
Comment (ethn) [inline card/mod/bootstrap/set/abstract/bootswatch_theme.rb]: how bad is it that this line isn't in 0.5?

File: card/mod/basic_types/set/type/json.rb [hunk 2]
```diff
@@ -1,5 +1,5 @@
 event :validate_json, :validate, on: :save, changed: :content do
-  check_json_syntax
+  check_json_syntax if content.present?
```
Comment (ethn) [inline card/mod/basic_types/set/type/json.rb]: ok for now.  wondering if we shouldn't have a default rule setting content to {}?

File: card/mod/bootstrap/set/abstract/bootswatch_theme.rb (hunk 1 above)
Comment (xithan) [reply card/mod/bootstrap/set/abstract/bootswatch_theme.rb]: no problem. we need that only for #248

File: card/mod/basic_types/set/type/json.rb (hunk 2 above)
Comment (xithan) [reply card/mod/basic_types/set/type/json.rb]: I agree in principal but "" is only a bad query, "{}" searches for all cards. That could cause a lot of performance trouble.

**Pull request merged**